### PR TITLE
#366 Initial support for DFT_P03

### PR DESCRIPTION
**Pull request by cragun47**
Signed-off-by: Brian Cragun <cragun@us.ibm.com>

This adds initial support for DFT_P03 and associated testing.

### CODE REVIEW

File: src/main/resources/config.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 base.path.resource=
-supported.hl7.messages=ADT_A01, ADT_A08, ADT_A34, ADT_A40, MDM_T02, MDM_T06, ORM_O01, OMP_O09, ORU_R01, PPR_PC1, RDE_O11, RDE_O25, VXU_V04
+supported.hl7.messages=ADT_A01, ADT_A08, ADT_A34, ADT_A40, DFT_P03, MDM_T02, MDM_T06, ORM_O01, OMP_O09, ORU_R01, PPR_PC1, RDE_O11, RDE_O25, VXU_V04
```
Comment (klwhaley) [inline src/main/resources/config.properties:2]: looks good, just a reminder can you post on dev ops to have them add this t the configuration properties so it gets deployed correctly for sue

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java [hunk 2]
```diff
@@ -36,9 +38,8 @@
...
     @Test
     void testBasicInsuranceCoverageFields() throws IOException {
-        // Currently only tests limited items, other fields to be added
-
-        String hl7message = "MSH|^~\\&|||||20151008111200||ADT^A01^ADT_A01|MSGID000001|T|2.6|||||||||\n"
+        // Tests fields listed below.  
+        String hl7message = "MSH|^~\\&|||||20151008111200||DFT^P03^DFT_P03|MSGID000001|T|2.6|||||||||\n"
                 + "EVN||20210407191342||||||\n"
                 + "PID|||MR1^^^XYZ^MR||DOE^JANE^|||F||||||||||||||||||||||\n"
                 + "PV1||I||||||||||||||||||||||||||||||||||||||||||\n"
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:45]: given FT1 is required, i know we don't convert it, but do you think we should have a simple version of it in the message?

File: src/main/resources/config.properties (hunk 1 above)
Comment (klwhaley) [inline src/main/resources/config.properties:2]: Also need to update the readme to include this new message type (and maybe a comment that FT1 is not supported for conversion)

File: src/main/resources/hl7/message/DFT_P03.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    segment: PV1
+    resourcePath: resource/Encounter
+    repeats: false
+    isReferenced: true
+    additionalSegments:
+      - PV2
+      - EVN
+      - MSH
+      - DG1
+
```
Comment (klwhaley) [inline src/main/resources/hl7/message/DFT_P03.yml:39]: ORC/OBR/OBX/NTE are supported in DFT_P03 messages...feel like we need something for those

File: src/main/resources/hl7/message/DFT_P03.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#
+# (C) Copyright IBM Corp. 2020, 2021
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# FHIR Resources to extract from DFT_P03 message
+# 
+################################
+# NOTE: FT1 is not currently utilized. FT1 records are ignored.
+---
```
Comment (klwhaley) [inline src/main/resources/hl7/message/DFT_P03.yml:10]: Need to add a new test file to test the groups/resource creation:

/hl7v2-fhir-converter/src/test/java/io/github/linuxforhealth/hl7/message/DFTMessageTest.java or something like that

**Review by klwhaley — COMMENTED** [review]
See comments in review

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java (hunk 2 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7FinancialInsuranceTest.java:45]: I have added it to the test messages and made a note that it is currently not used (ignored).  

The only concern I have about doing so, is that it means our tests are now hiding the fact that DFT_P03 _should_ but _apparently does not_ require FT1.

File: src/main/resources/config.properties (hunk 1 above)
Comment (cragun47) [reply src/main/resources/config.properties:2]: Done.

File: src/main/resources/hl7/message/DFT_P03.yml (hunk 3 above)
Comment (cragun47) [reply src/main/resources/hl7/message/DFT_P03.yml:39]: I can add them.  Are you wanting tests constructed for them?  @klwhaley  ?

File: src/main/resources/hl7/message/DFT_P03.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  - resourceName: Coverage
+    segment: .IN1
+    group: INSURANCE
+    resourcePath: resource/Coverage
+    repeats: true
+    additionalSegments:
+      - MSH
+      - PID
+      - PV1
```
[the remark this answers is not in the document]
Comment (cragun47) [reply src/main/resources/hl7/message/DFT_P03.yml:48]: True.  Removed.

File: src/main/resources/hl7/message/DFT_P03.yml (hunk 3 above)
Comment (cragun47) [reply src/main/resources/hl7/message/DFT_P03.yml:39]: Adding simplified presence tests to Hl7DFTMessageTest

File: src/main/resources/hl7/message/DFT_P03.yml (hunk 4 above)
Comment (cragun47) [reply src/main/resources/hl7/message/DFT_P03.yml:10]: Done.  Created `testResourceCreationFromDFT` which tests ServiceRequest and Observation (and associated NTEs) from added support for ORC/OBR/OBX/NTE in `DFT_P03.yml`

File: README.md [hunk 6]
```diff
@@ -9,6 +9,7 @@ The converter supports the following message types/events:
 * ADT_A08 - Patient Administration: Update Patient Information
 * ADT_A34 - Patient Administration: Merge Patient Information - Patient ID Only
 * ADT_A40 - Patient Administration: Merge Patient - Patient Identifier List
+* DFT_P03 - Post Detail Financial Transaction (does not utilize FT1)
```
Comment (klwhaley) [inline README.md:12]: should we say convert FT1? i'm okay either way

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7DFTMessageTest.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
+/*
+ * (C) Copyright IBM Corp. 2021
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/message/Hl7DFTMessageTest.java:2]: 2022

**Review by klwhaley — APPROVED** [review]
THANK YOU!

File: README.md (hunk 6 above)
Comment (cragun47) [reply README.md:12]: convert FT1 sounds better.

Comment (cragun47) [reply README.md:12]: done

File: src/test/java/io/github/linuxforhealth/hl7/message/Hl7DFTMessageTest.java (hunk 7 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/message/Hl7DFTMessageTest.java:2]: Done

File: src/main/resources/config.properties (hunk 1 above)
Comment (cragun47) [reply src/main/resources/config.properties:2]: Done

**Pull request merged**